### PR TITLE
Charlesmchen/gather analytics event names

### PR DESCRIPTION
**Pull request by charlesmchen**
PTAL @michaelkirk

### CODE REVIEW

**Comment by charlesmchen** [comment]
@michaelkirk 95% of the changes in this PR were code generated by a python script.  I didn't bother cleaning up that python script much since I'm not sure if we're going to use it going forward.  We could - like the l10n scripts it normalizes, sorts, and ensures a consistent mapping between the "event name" and the method name used to access it.

File: SignalServiceKit/src/Util/OWSAnalyticsEvents.h [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+//  Copyright (c) 2017 Open Whisper Systems. All rights reserved.
+//
+
+NS_ASSUME_NONNULL_BEGIN
+
+@interface OWSAnalyticsEvents : NSObject
+
+- (instancetype)init NS_UNAVAILABLE;
+
+#pragma mark - Code Generation Marker
```
Comment (michaelkirk) [inline SignalServiceKit/src/Util/OWSAnalyticsEvents.h]: Maybe a comment to tie this back to the python script to talk about how this code was generated.

File: SignalServiceKit/Utilities/extract_analytics_event_names.py [hunk 2]
```diff
@@ -0,0 +1,368 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+import os
```
Comment (michaelkirk) [inline SignalServiceKit/Utilities/extract_analytics_event_names.py]: maybe a line or two comment explaining what this was used for.

**Comment by michaelkirk** [comment]
This will conflict with https://github.com/WhisperSystems/Signal-iOS/commit/898f122d2b02d8eb1332a62f9965b4b0cb65aa90 which modified some event names - so hopefully you just need to re-run that script.

File: SignalServiceKit/src/Util/OWSAnalyticsEvents.h (hunk 1 above)
Comment (charlesmchen) [inline SignalServiceKit/src/Util/OWSAnalyticsEvents.h]: Done.

File: SignalServiceKit/Utilities/extract_analytics_event_names.py (hunk 2 above)
Comment (charlesmchen) [inline SignalServiceKit/Utilities/extract_analytics_event_names.py]: Done. On reflection, I think we should continue to use this script as we add more analytics events.

**Comment by charlesmchen** [comment]
> This will conflict with 898f122 which modified some event names - so hopefully you just need to re-run that script.

Yup, no problem.  Re-ran the script.

**Pull request merged**